[docs] Update Clerk guide to match the current Clerk Expo quickstart (#48308)
expo/expo: docs/pages/guides/using-clerk.mdx

<TabsGroup>

[Clerk](https://clerk.com/expo-authentication) is an authentication and user management platform that provides sign-up, sign-in, multi-factor authentication, social sign-in, organizations, and a hosted user database. The [`@clerk/expo`](https://www.npmjs.com/package/@clerk/expo) SDK gives you React hooks, control components, hosted authentication, and prebuilt native UI components that render with Jetpack Compose on Android and SwiftUI on iOS.

> This guide shows you how to install `@clerk/expo`, wrap your app in `<ClerkProvider>`, and choose the integration approach that fits your project. It targets `@clerk/expo` Core 3 (the 3.x release line), which supports Expo SDK 53, 54, and 55.

Reviewer: ⚠️ **correctness** `mdx-components.md#variables`

`@clerk/expo/hosted-auth` and `@clerk/expo-google-signin` only exist in `@clerk/expo` 4.x, so the hosted authentication sections added here do not work on the 3.x line this sentence targets: 3.7.8, the last 3.x release, exports no `./hosted-auth` subpath. 4.1.2 declares `expo: >=54 <58`, which also makes "Expo SDK 53, 54, and 55" wrong, along with the `expo: >=53 <56` peer dependency in the prerequisites below. Retarget the page to 4.x.

```suggestion
This guide shows you how to install `@clerk/expo`, wrap your app in `<ClerkProvider>`, and choose the integration approach that fits your project. It targets `@clerk/expo` 4.x, which supports Expo SDK 54 and later.
```

<!-- docs-review -->

| --------------------- | --------------------------------------------------------------------------------------------- | --------------- | -------------------------------------------------------------- |
| Hosted authentication | A button that opens Clerk's Account Portal in a browser authentication session                | <YesIcon />     | The fastest setup, with every method enabled in your dashboard |
| Native UI components  | Drop in `<AuthView />`, `<UserButton />`, and `<UserProfileView />` from `@clerk/expo/native` | <NoIcon />      | A complete native sign-in and account management UI            |
| Custom flow           | Your own React Native screens that call hooks such as `useSignUp()` and `useSignIn()`         | <YesIcon />     | Maximum UI control                                             |

> **info** The native UI components in `@clerk/expo/native` are currently in beta. They render with Jetpack Compose on Android and SwiftUI on iOS and they synchronize the signed-in session back to the JavaScript SDK so all `@clerk/expo` hooks (such as `useAuth()` and `useUser()`) stay in sync.

> The custom flow approach runs in Expo Go as-is. Adding native Sign in with Google or Sign in with Apple buttons to it requires a development build because those hooks use native modules.

Reviewer: ✏️ **style** `review-ai-text.md#t5-restatement`

The table directly above already marks the custom flow as running in Expo Go, and the "Use a development build for native features" requirement below already says the native sign-in hooks need a development build. Cut this paragraph.

<!-- docs-review -->

<Step label="2">

### Verify the config plugins

> Expo automatically adds the required config plugins to your [app config](/workflow/configuration/) when you install the packages. Verify that `@clerk/expo` and `expo-secure-store` appear in the `plugins` array:

Reviewer: ⚠️ **correctness** `review-ai-text.md#t7-overclaiming`

`npx expo install` only writes the `plugins` array for a static app config. A project on **app.config.js** or **app.config.ts** gets "Cannot automatically write to dynamic config" instead and has to add the plugins by hand, so this line and the "Verify the config plugins" heading leave those readers without the step. Keep the instruction to add them and mention the automatic case as the shortcut.

```suggestion
Add `@clerk/expo` and `expo-secure-store` to the `plugins` array in your [app config](/workflow/configuration/). If your project uses a static **app.json** and you installed the packages with `npx expo install`, Expo has already added them:
```

<!-- docs-review -->

```json app.json
{
  "expo": {
    "plugins": ["expo-secure-store", "@clerk/expo"]
  }
}
```

The `@clerk/expo` plugin adds the Apple Sign In entitlement (disable it with the `appleSignIn: false` plugin option if your app doesn't use it), registers the Android intent filter for the hosted authentication callback, and applies the Android packaging fixes required by the underlying `clerk-android` SDK. If you use the native Sign in with Google buttons, also add the `@clerk/expo-google-signin` plugin alongside it.

> Hosted authentication derives its default callback from the `ios.bundleIdentifier` and `android.package` values in your app config. Before you create a production build, add the app on the [Native applications](https://dashboard.clerk.com/last-active?path=native-applications) page in the Clerk Dashboard with the same iOS bundle identifier and Android package name, since production instances validate the callback against the registered values.

Reviewer: ✏️ **style** `terminology.md#platform-order`

Flowing prose orders platforms Android, iOS, and web, so swap both pairs on this line. The same swap applies to "your iOS bundle identifier or Android package name" in the hosted authentication tab.

```suggestion
Hosted authentication derives its default callback from the `android.package` and `ios.bundleIdentifier` values in your app config. Before you create a production build, add the app on the [Native applications](https://dashboard.clerk.com/last-active?path=native-applications) page in the Clerk Dashboard with the same Android package name and iOS bundle identifier, since production instances validate the callback against the registered values.
```

<!-- docs-review -->

import { useState } from 'react';
import { ActivityIndicator, Button, Modal, View } from 'react-native';

export default function MainScreen() {
  const { isLoaded, isSignedIn } = useAuth({ treatPendingAsSignedOut: false });
  const [isAuthOpen, setIsAuthOpen] = useState(false);

  if (!isLoaded) {
    return <ActivityIndicator size="large" />;
  }

  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      {isSignedIn ? <UserButton /> : <Button title="Sign up" onPress={() => setIsAuthOpen(true)} />}
      <Modal
        animationType="slide"
        visible={isAuthOpen}
        presentationStyle="pageSheet"
        onRequestClose={() => setIsAuthOpen(false)}>
        <AuthView onDismiss={() => setIsAuthOpen(false)} />
      </Modal>
    </View>
  );
}
```

After the user signs in, the native session is synchronized back to the JavaScript SDK, so `useAuth()` and `useUser()` reflect the signed-in state. Pass `treatPendingAsSignedOut: false` to `useAuth()` so pending [session tasks](https://clerk.com/docs/guides/development/custom-flows/authentication/session-tasks) are not treated as signed out.

> > **warning** Keep the `<Modal>` that contains `<AuthView />` mounted at the same level as your signed-in and signed-out content. Don't render the modal only inside signed-out content, because auth state can change before required session tasks are finished and unmount the modal too early.

Reviewer: ✏️ **style** `prose.md#voice-and-tone`

The second sentence makes "auth state" the subject of "unmount", so it reads as though the state unmounts the modal. Name the actual cause: the state change flips your conditional render, which unmounts the modal while session tasks are still pending.

```suggestion
> **warning** Keep the `<Modal>` that contains `<AuthView />` mounted at the same level as your signed-in and signed-out content. If you render it only inside signed-out content, auth state can change while session tasks are still pending, and your conditional render unmounts the modal too early.
```

<!-- docs-review -->

`<AuthView />` accepts `mode="signIn" | "signUp" | "signInOrUp"` (the default), an `isDismissible` boolean that controls the native dismiss button, and an `onDismiss` callback. When users must authenticate before continuing, render it as a full-screen view with `isDismissible={false}` instead of in a modal.

`<UserButton />` takes no props. It displays the signed-in user's profile image or initials and opens the native `<UserProfileView />` when tapped, where users can manage personal information, security settings, and sign out.

> `<AuthView />` automatically shows sign-in buttons for any social connections enabled in the Clerk Dashboard and handles the flows internally, so you don't need `expo-crypto` or the native sign-in hooks. Native OAuth still requires credential setup in the Clerk Dashboard and provider consoles, otherwise the buttons appear but fail when tapped. Follow the Clerk Sign in with Google and Sign in with Apple guides linked at the bottom of this page.

Reviewer: ✏️ **style** `prose.md#sentence-construction`

"provider consoles, otherwise the buttons appear but fail when tapped" is a comma splice. Start a new sentence at "Otherwise".

```suggestion
`<AuthView />` automatically shows sign-in buttons for any social connections enabled in the Clerk Dashboard and handles the flows internally, so you don't need `expo-crypto` or the native sign-in hooks. Native OAuth still requires credential setup in the Clerk Dashboard and provider consoles. Otherwise, the buttons appear but fail when tapped. Follow the Clerk Sign in with Google and Sign in with Apple guides linked at the bottom of this page.
```

<!-- docs-review -->

      <TextInput
        value={password}
        placeholder="Enter password"
        secureTextEntry
        onChangeText={setPassword}
      />
      <Button title="Sign up" onPress={handleSignUp} />
      {/* Required for sign-up flows on Expo web. Clerk skips the browser CAPTCHA on Android and iOS */}
      <View nativeID="clerk-captcha" />
    </View>
  );
}
```

In Core 3, methods such as `signUp.password()` and `signUp.verifications.verifyEmailCode()` return `{ error }` instead of throwing for validation errors. When verification completes the sign-up, `signUp.finalize()` converts it into an active session and updates `useAuth()` with the signed-in state.

The corresponding sign-in flow uses `useSignIn()`, where `finalize()` accepts a `navigate` callback so you can handle [session tasks](https://clerk.com/docs/guides/development/custom-flows/authentication/session-tasks) before redirecting:

```tsx
const { error } = await signIn.password({ emailAddress, password });
if (error) return;

if (signIn.status === 'complete') {
  await signIn.finalize({
    navigate: ({ session, decorateUrl }) => {
      if (session?.currentTask) return; // let the session task layer handle it

>       router.replace(decorateUrl('/') as Href);

Reviewer: ⚠️ **correctness** `prose.md#core-principles`

`router` and `Href` are not defined anywhere on this page anymore, because the example that imported `useRouter` and `type Href` from `expo-router` is the one this PR replaced. Add the import to this snippet, or move the `navigate` callback into prose.

<!-- docs-review -->

    },
  });
}
```

> To add native Sign in with Google and Sign in with Apple buttons to your custom screens, use the `useSignInWithGoogle()` hook from `@clerk/expo/google` and the `useSignInWithApple()` hook from `@clerk/expo/apple`. Both return a start method (`startGoogleAuthenticationFlow()` and `startAppleAuthenticationFlow()`) that resolves with `{ createdSessionId, setActive }`. These hooks use native modules, so they require a development build and the packages from the install step, and `useSignInWithGoogle()` also needs the `@clerk/expo-google-signin` config plugin. On iOS, App Store Guideline 4.8 requires that any app offering third-party social sign-in must also offer Sign in with Apple. Follow the Clerk guides linked at the bottom of this page to register your credentials in the Clerk Dashboard and the provider consoles.

Reviewer: ✏️ **style** `prose.md#voice-and-tone`

"These hooks use native modules, so they require a development build and the packages from the install step, and `useSignInWithGoogle()` also needs the `@clerk/expo-google-signin` config plugin" chains three requirements into one sentence. Split them out and break this 100-word paragraph in two.

```suggestion
To add native Sign in with Google and Sign in with Apple buttons to your custom screens, use the `useSignInWithGoogle()` hook from `@clerk/expo/google` and the `useSignInWithApple()` hook from `@clerk/expo/apple`. Both return a start method (`startGoogleAuthenticationFlow()` and `startAppleAuthenticationFlow()`) that resolves with `{ createdSessionId, setActive }`.

Both hooks use native modules, so they require a development build and the packages from the install step. `useSignInWithGoogle()` also needs the `@clerk/expo-google-signin` config plugin. On iOS, App Store Guideline 4.8 requires that any app offering third-party social sign-in must also offer Sign in with Apple. Follow the Clerk guides linked at the bottom of this page to register your credentials in the Clerk Dashboard and the provider consoles.
```

<!-- docs-review -->

## Run the app

> <Tabs tabs={['Android', 'iOS', 'Expo Go']}>

Reviewer: ⚠️ **correctness** `mdx-components.md#tabs-and-tab`

The `<TabsGroup>` wrapper shares one selected index across both tab sets on this page, so index 0 pairs "Hosted authentication" with "Android" and a reader on hosted authentication, which the table marks as running in Expo Go, is shown `npx expo run:android`. The previous labels kept index 2 as "JavaScript-only (Expo Go)" in both sets, so that pairing held. Move the Expo Go tab and its `<Tab>` body to the first position so it lines up with "Hosted authentication".

<!-- docs-review -->
